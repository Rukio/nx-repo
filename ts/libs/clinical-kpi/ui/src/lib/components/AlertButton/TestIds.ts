export const ALERT_BUTTON_TEST_IDS = {
  TEXT: 'alert-button-text',
  BUTTON: 'alert-button-button',
};
