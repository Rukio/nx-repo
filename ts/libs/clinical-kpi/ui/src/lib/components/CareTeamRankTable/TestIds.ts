export const CARE_TEAM_RANK_TABLE_TEST_IDS = {
  TABLE: 'care-team-rank-table',
  getRankTestId: (rank?: number) => `care-team-table-rank-${rank}`,
  getProviderNameTestId: (rank?: number) =>
    `care-team-rank-table-provider-name-${rank}`,
  getValueTestId: (rank?: number) => `care-team-rank-table-value-${rank}`,
  getValueChangeTestId: (rank?: number) =>
    `care-team-rank-table-value-change-${rank}`,
  getPlaceholderTestId: (rank?: number) =>
    `care-team-rank-table-placeholder-${rank}`,
};

export const CARE_TEAM_PLACEHOLDER_ROW_RANKS = [1, 2, 3, 4, 5, 6, 7];
