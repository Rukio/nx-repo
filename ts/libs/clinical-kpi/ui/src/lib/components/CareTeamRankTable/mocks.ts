export type LeaderHubRankTableRow = {
  id: number;
  rank: number;
  name: string;
  value?: number;
  valueChange: number;
  position?: string;
};

export const careTeamRankingsMock: LeaderHubRankTableRow[] = [
  {
    id: 1,
    name: 'Luigi Martiniello',
    rank: 1,
    value: 5.67,
    valueChange: -11.67,
    position: 'DHMT',
  },
  {
    id: 2,
    name: 'Connie Garcia',
    rank: 2,
    value: 6,
    valueChange: -0.02,
    position: 'APP',
  },
  {
    id: 3,
    name: 'Steve Tomis',
    rank: 3,
    value: 6.17,
    valueChange: 1.67,
    position: 'APP',
  },
  {
    id: 4,
    name: 'Veerle de Bree',
    rank: 4,
    value: 7.02,
    valueChange: 0,
    position: 'APP',
  },
  {
    id: 5,
    name: '',
    rank: 5,
    value: 7.5,
    valueChange: 0.17,
    position: 'DHMT',
  },
];
