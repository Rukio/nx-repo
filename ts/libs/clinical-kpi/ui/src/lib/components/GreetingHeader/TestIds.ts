export const GREETING_HEADER_TEST_IDS = {
  GREETING_TEXT: 'header-greeting-text',
  LAST_UPDATED_TEXT: 'header-last-updated-text',
  VISITS_COMPLETED_TEXT: 'header-visits-completed-text',
  PERFORMANCE_TEXT: 'header-performance-text',
  LEARN_MORE_LINK: 'header-learn-more-link',
  LAST_UPDATED_PLACEHOLDER: 'header-last-updated-placeholder',
  VISITS_COMPLETED_PLACEHOLDER: 'header-visits-completed-placeholder',
  STATION_LINK: 'header-station-dashboard-link',
  STATION_LINK_LABEL_PREFIX: 'header-station-dashboard-link-label-prefix',
};
