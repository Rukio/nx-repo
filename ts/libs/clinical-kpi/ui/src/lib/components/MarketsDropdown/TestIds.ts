export const MARKETS_DROPDOWN_TEST_IDS = {
  LABEL: 'markets-dropdown-label',
  SELECT: 'markets-dropdown-select',
  SELECT_ITEM: (id: string) => `markets-dropdown-select-item-${id}`,
};
