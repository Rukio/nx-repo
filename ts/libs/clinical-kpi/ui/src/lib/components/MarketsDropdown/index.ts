export { default as MarketsDropdown } from './MarketsDropdown';
export { MARKETS_DROPDOWN_TEST_IDS } from './TestIds';
