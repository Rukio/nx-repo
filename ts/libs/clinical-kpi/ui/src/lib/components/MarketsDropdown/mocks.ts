import { Market } from './MarketsDropdown';

export const marketsListMock: Market[] = [
  {
    id: '176',
    name: 'Boise',
  },
  {
    id: '198',
    name: 'Columbus',
  },
];
