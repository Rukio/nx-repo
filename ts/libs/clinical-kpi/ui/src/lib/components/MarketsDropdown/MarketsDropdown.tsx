import {
  Select,
  MenuItem,
  FormControl,
  InputLabel,
  SelectChangeEvent,
  makeSxStyles,
} from '@*company-data-covered*/design-system';

import { MARKETS_DROPDOWN_TEST_IDS } from './TestIds';

export interface Market {
  name: string;
  id: string;
}

export interface MarketsDropdownProps {
  selectedMarketId?: string;
  onMarketChange?: (marketId: string) => void;
  markets: Market[];
}

const makeStyles = () =>
  makeSxStyles({
    selectWrapper: {
      minWidth: '220px',
    },
  });

const MarketsDropdown: React.FC<MarketsDropdownProps> = ({
  selectedMarketId,
  onMarketChange,
  markets,
}) => {
  const styles = makeStyles();

  const handleMarketChange = (event: SelectChangeEvent) => {
    onMarketChange?.(event.target.value);
  };

  return (
    <FormControl sx={styles.selectWrapper} size="medium">
      <InputLabel data-testid={MARKETS_DROPDOWN_TEST_IDS.LABEL}>
        Market
      </InputLabel>
      <Select
        data-testid={MARKETS_DROPDOWN_TEST_IDS.SELECT}
        variant="outlined"
        size="medium"
        label="Market"
        onChange={handleMarketChange}
        value={selectedMarketId ?? ''}
      >
        {markets.map(({ id, name }) => (
          <MenuItem
            key={id}
            data-testid={MARKETS_DROPDOWN_TEST_IDS.SELECT_ITEM(id)}
            value={id}
          >
            {name}
          </MenuItem>
        ))}
      </Select>
    </FormControl>
  );
};

export default MarketsDropdown;
