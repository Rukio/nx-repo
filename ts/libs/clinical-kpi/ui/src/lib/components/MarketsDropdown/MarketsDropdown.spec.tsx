import { render, screen } from '../../../testUtils';
import MarketsDropdown, {
  Market,
  MarketsDropdownProps,
} from './MarketsDropdown';
import { MARKETS_DROPDOWN_TEST_IDS } from './TestIds';
import { marketsListMock } from './mocks';

const onMarketChange = jest.fn().mockImplementation((id: string) => undefined);

const setup = (markets: MarketsDropdownProps['markets'] = marketsListMock) => {
  const getMarketDropdown = () =>
    screen.getByRole('button', {
      ...screen.getByTestId(MARKETS_DROPDOWN_TEST_IDS.SELECT),
      expanded: false,
    });

  const getMarketDropdownItem = (marketId: Market['id']) =>
    screen.getByTestId(MARKETS_DROPDOWN_TEST_IDS.SELECT_ITEM(marketId));

  return {
    ...render(
      <MarketsDropdown onMarketChange={onMarketChange} markets={markets} />
    ),
    getMarketDropdown,
    getMarketDropdownItem,
  };
};

describe('MarketsDropdown', () => {
  it('should render correctly', async () => {
    const { user, baseElement, getMarketDropdownItem, getMarketDropdown } =
      setup();

    const selectElement = getMarketDropdown();

    expect(
      screen.getByTestId(MARKETS_DROPDOWN_TEST_IDS.LABEL).textContent
    ).toContain('Market');

    await user.click(selectElement);

    marketsListMock.forEach((market) => {
      expect(getMarketDropdownItem(market.id)).toBeVisible();
    });

    expect(baseElement).toMatchSnapshot();
  });

  it('should call onMarketChange function when select new market', async () => {
    const { user, getMarketDropdownItem, getMarketDropdown } = setup();

    const selectElement = getMarketDropdown();

    await user.click(selectElement);

    const marketToSelect = marketsListMock[0];
    const marketDropdownItemToSelect = getMarketDropdownItem(marketToSelect.id);

    await user.click(marketDropdownItemToSelect);

    expect(onMarketChange).toBeCalledTimes(1);
    expect(onMarketChange).toBeCalledWith(marketToSelect.id);
  });
});
