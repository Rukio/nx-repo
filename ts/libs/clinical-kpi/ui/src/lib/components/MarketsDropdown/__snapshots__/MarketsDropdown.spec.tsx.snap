// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketsDropdown should render correctly 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  min-width: 220px;
}

.emotion-1 {
  color: rgba(0, 0, 0, 0.6);
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  padding: 0;
  position: relative;
  display: block;
  transform-origin: top left;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: calc(133% - 24px);
  position: absolute;
  left: 0;
  top: 0;
  -webkit-transform: translate(14px, -9px) scale(0.75);
  -moz-transform: translate(14px, -9px) scale(0.75);
  -ms-transform: translate(14px, -9px) scale(0.75);
  transform: translate(14px, -9px) scale(0.75);
  -webkit-transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,-webkit-transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: color 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,transform 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms,max-width 200ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  z-index: 1;
  pointer-events: auto;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.emotion-1.Mui-focused {
  color: #1976d2;
}

.emotion-1.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
}

.emotion-1.Mui-error {
  color: #d32f2f;
}

.emotion-2 {
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  letter-spacing: 0.00938em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  border-radius: 4px;
}

.emotion-2.Mui-disabled {
  color: rgba(0, 0, 0, 0.38);
  cursor: default;
}

.emotion-2:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-2:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-2.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #1976d2;
  border-width: 2px;
}

.emotion-2.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-2.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-3 {
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 4px;
  cursor: pointer;
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 4px 0 5px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding: 16.5px 14px;
}

.emotion-3:focus {
  border-radius: 4px;
}

.emotion-3::-ms-expand {
  display: none;
}

.emotion-3.Mui-disabled {
  cursor: default;
}

.emotion-3[multiple] {
  height: auto;
}

.emotion-3:not([multiple]) option,
.emotion-3:not([multiple]) optgroup {
  background-color: #fff;
}

.emotion-3.emotion-3.emotion-3 {
  padding-right: 32px;
}

.emotion-3.MuiSelect-select {
  height: auto;
  min-height: 1.4375em;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: rgba(0, 0, 0, 0.38);
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-3:-webkit-autofill {
  border-radius: inherit;
}

.emotion-4 {
  bottom: 0;
  left: 0;
  position: absolute;
  opacity: 0;
  pointer-events: none;
  width: 100%;
  box-sizing: border-box;
}

.emotion-5 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  position: absolute;
  right: 7px;
  top: calc(50% - .5em);
  pointer-events: none;
  color: rgba(0, 0, 0, 0.54);
  -webkit-transform: rotate(180deg);
  -moz-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
}

.emotion-5.Mui-disabled {
  color: rgba(0, 0, 0, 0.26);
}

.emotion-6 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-7 {
  float: unset;
  width: auto;
  overflow: hidden;
  display: block;
  padding: 0;
  height: 11px;
  font-size: 0.75em;
  visibility: hidden;
  max-width: 100%;
  -webkit-transition: max-width 100ms cubic-bezier(0.0, 0, 0.2, 1) 50ms;
  transition: max-width 100ms cubic-bezier(0.0, 0, 0.2, 1) 50ms;
  white-space: nowrap;
}

.emotion-7>span {
  padding-left: 5px;
  padding-right: 5px;
  display: inline-block;
  opacity: 0;
  visibility: visible;
}

.emotion-8 {
  position: fixed;
  z-index: 1300;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
}

.emotion-9 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  right: 0;
  bottom: 0;
  top: 0;
  left: 0;
  background-color: transparent;
  -webkit-tap-highlight-color: transparent;
  z-index: -1;
}

.emotion-10 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  max-height: calc(100% - 96px);
  -webkit-overflow-scrolling: touch;
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12);
  position: absolute;
  overflow-y: auto;
  overflow-x: hidden;
  min-width: 16px;
  min-height: 16px;
  max-width: calc(100% - 32px);
  max-height: calc(100% - 32px);
  outline: 0;
}

.emotion-11 {
  list-style: none;
  margin: 0;
  padding: 0;
  position: relative;
  padding-top: 8px;
  padding-bottom: 8px;
  outline: 0;
}

.emotion-12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
  min-height: 48px;
  padding-top: 6px;
  padding-bottom: 6px;
  box-sizing: border-box;
  white-space: nowrap;
  padding-left: 16px;
  padding-right: 16px;
}

.emotion-12::-moz-focus-inner {
  border-style: none;
}

.emotion-12.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-12 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-12:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-12:hover {
    background-color: transparent;
  }
}

.emotion-12.Mui-selected {
  background-color: rgba(25, 118, 210, 0.08);
}

.emotion-12.Mui-selected.Mui-focusVisible {
  background-color: rgba(25, 118, 210, 0.2);
}

.emotion-12.Mui-selected:hover {
  background-color: rgba(25, 118, 210, 0.12);
}

@media (hover: none) {
  .emotion-12.Mui-selected:hover {
    background-color: rgba(25, 118, 210, 0.08);
  }
}

.emotion-12.Mui-focusVisible {
  background-color: rgba(0, 0, 0, 0.12);
}

.emotion-12.Mui-disabled {
  opacity: 0.38;
}

.emotion-12+.MuiDivider-root {
  margin-top: 8px;
  margin-bottom: 8px;
}

.emotion-12+.MuiDivider-inset {
  margin-left: 52px;
}

.emotion-12 .MuiListItemText-root {
  margin-top: 0;
  margin-bottom: 0;
}

.emotion-12 .MuiListItemText-inset {
  padding-left: 36px;
}

.emotion-12 .MuiListItemIcon-root {
  min-width: 36px;
}

@media (min-width:600px) {
  .emotion-12 {
    min-height: auto;
  }
}

.emotion-13 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

<body
  style="padding-right: 1024px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  >
    <div
      class="MuiFormControl-root emotion-0"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-colorPrimary Mui-focused MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined emotion-1"
        data-shrink="true"
        data-testid="markets-dropdown-label"
      >
        Market
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl  emotion-2"
        data-testid="markets-dropdown-select"
      >
        <div
          aria-expanded="true"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input emotion-3"
          role="button"
          tabindex="0"
        >
          <span
            class="notranslate"
          >
            ​
          </span>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput emotion-4"
          tabindex="-1"
          value=""
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined MuiSelect-iconOpen emotion-5"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline emotion-6"
        >
          <legend
            class="emotion-7"
          >
            <span>
              Market
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </div>
  <div
    class="MuiPopover-root MuiMenu-root MuiModal-root emotion-8"
    id="menu-"
    role="presentation"
  >
    <div
      aria-hidden="true"
      class="MuiBackdrop-root MuiBackdrop-invisible emotion-9"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    />
    <div
      data-testid="sentinelStart"
      tabindex="0"
    />
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiPaper-root MuiMenu-paper MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiPopover-paper emotion-10"
      style="opacity: 1; transform: none; min-width: 0; transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;"
      tabindex="-1"
    >
      <ul
        class="MuiList-root MuiList-padding MuiMenu-list emotion-11"
        role="listbox"
        tabindex="-1"
      >
        <li
          aria-selected="false"
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-focusVisible MuiMenuItem-root MuiMenuItem-gutters emotion-12"
          data-testid="markets-dropdown-select-item-176"
          data-value="176"
          role="option"
          tabindex="0"
        >
          Boise
          <span
            class="MuiTouchRipple-root emotion-13"
          />
        </li>
        <li
          aria-selected="false"
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters emotion-12"
          data-testid="markets-dropdown-select-item-198"
          data-value="198"
          role="option"
          tabindex="-1"
        >
          Columbus
          <span
            class="MuiTouchRipple-root emotion-13"
          />
        </li>
      </ul>
    </div>
    <div
      data-testid="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;
